### ISSUE TITLE
Is there a way to run Sahriswiki without a proxy pass?

### ISSUE THREAD

**Issue by PeteBleackley** [prologic/sahriswiki#8]
I am trying to set up a Sahriswiki instance on a shared Dreamhost account. Because it's a shared account, I can't access the Apache config file, and whenever I ask Dreamhost support to do anything for me, they refuse and tell me to upgrade my account (which I can't do, because it's a shared account). Is there a way to run Sahriswiki without setting up a proxy?

**Comment by prologic** [comment]
Does Dreamhost support running WSGI apps?

James Mills / prologic

E: prologic@shortcircuit.net.au
W: prologic.shortcircuit.net.au

**Comment by PeteBleackley** [comment]
Apparently it does.

**Comment by prologic** [comment]
Then I'll provide an option to run this as a WSGI app shortly. THanks for
the feedback!

James Mills / prologic

E: prologic@shortcircuit.net.au
W: prologic.shortcircuit.net.au

**Comment by PeteBleackley** [comment]
Any progress on this?

**Comment by prologic** [comment]
I'm sorry I did try to get this working as a plain 'ol wsgi app but
unfortunately it turned out to be a lot more effort than I thought. I
haven't gotten back to this yet due to time constraints. Sorry :/

James Mills / prologic

E: prologic@shortcircuit.net.au
W: prologic.shortcircuit.net.au

**Comment by PeteBleackley** [comment]
Can I help with this at all?

**Comment by prologic** [comment]
Yes you can! Let me put up the code so far in a branch. There are some
issues that need to be worked out; but it's dosable.

James Mills / prologic

E: prologic@shortcircuit.net.au
W: prologic.shortcircuit.net.au

**Comment by prologic** [comment]
I started this [branch](https://github.com/prologic/sahriswiki/tree/support_running_as_wsgi_app)

Hope you can help :)

**Comment by PeteBleackley** [comment]
It seems that there's something in the list returned by the application's **call** method that isn't a string, but I haven't been able to work out why. Will share to try to get more help

**Comment by PeteBleackley** [comment]
It seems this is a 2/3 issue with wsgiref. The application is returning unicode where wsgiref expects str.

**Comment by prologic** [comment]
Oh I see. Where is the bug? I presume in circuits.web.wsgi.Application?

Can you send a PR to fix this?

James Mills / prologic

E: prologic@shortcircuit.net.au
W: prologic.shortcircuit.net.au

**Comment by PeteBleackley** [comment]
I've managed to get an instance up and running at http://sources.conlang.org/

**Comment by prologic** [comment]
I WoW! Very nicely done! Do you mind sharing how you did this for the
benefit of others and/or submit any patches required?

cheers
James

James Mills / prologic

E: prologic@shortcircuit.net.au
W: prologic.shortcircuit.net.au

**Comment by PeteBleackley** [comment]
Here's the passenger_wsgi.py file I use on DreamHost

```
"""WSGI Compatibility

This module implements a WSGI Application that can be loded by mod_wsgi
or uwsgi.
"""

import os
import sys
sys.path.append('../sahriswiki-support_running_as_wsgi_app/sahriswiki')
from mercurial.ui import ui
from mercurial.hgweb import hgweb

from circuits import Debugger
from circuits.web import Sessions, Static
from circuits.web.wsgi import Application, Gateway

import sahriswiki
import sahriswiki.config
import sahriswiki.env
import sahriswiki.root
import sahriswiki.tools

config = sahriswiki.config.Config(open('wiki.config','r'))

environ = sahriswiki.env.Environment(config)

application = Application()

application += Debugger()
application += Sessions()
application += sahriswiki.root.Root(environ)
application += sahriswiki.tools.CacheControl(environ)
application += sahriswiki.tools.ErrorHandler(environ)

application += environ

if not config.get("disable-static"):
    application += Static(docroot=os.path.join(config.get("theme"), "htdocs"))

if not config.get("disable-hgweb"):
    baseui = ui()
    baseui.setconfig("web", "prefix", "/+hg")
    baseui.setconfig("web", "style", "gitweb")
    baseui.setconfig("web", "allow_push", "*")
    baseui.setconfig("web", "push_ssl", False)
    baseui.setconfig("web", "allow_archive", ["bz2", "gz", "zip"])
    baseui.setconfig("web", "description", config.get("description"))

    application += Gateway({
        "/+hg": hgweb(
            environ.storage.repo_path,
            config.get("name"),
            baseui
        )
    })

 if not config.get("disable-compression"):
    application += sahriswiki.tools.Compression(environ)
```

**Comment by prologic** [comment]
Do you want to submit a PR with a file like this in `scripts/` maybe?

**Comment by blade2005** [comment]
thank you thank you. was just trying to find out what  i needed to do for this exact thing.

**Comment by blade2005** [comment]
@PeteBleackley can you show me what you used for your config?
